fetch/cosmwasm_contract/src/contract.rs

```diff
@@ -817,91 +850,110 @@ fn send_tokens_from_contract<A: Api>(
     } else if amount > state.upper_swap_limit {
         Err(StdError::generic_err(ERR_SWAP_LIMITS_VIOLATED))
     } else {
-        Ok(HandleResponse::default())
+        Ok(Response::default())
     }
 }
 
-fn verify_aggregated_reverse_allowance(amount: Uint128, state: &State) -> HandleResult {
+fn verify_aggregated_reverse_allowance(
+    amount: Uint128,
+    state: &State,
+) -> Result<Response, StdError> {
     if state.reverse_aggregated_allowance < amount {
         Err(StdError::generic_err(ERR_RA_ALLOWANCE_EXCEEDED))
     } else {
-        Ok(HandleResponse::default())
+        Ok(Response::default())
     }
 }
 
-fn verify_refund_swap_id<S: Storage>(id: u64, storage: &S) -> HandleResult {
+fn verify_refund_swap_id(id: u64, storage: &dyn Storage) -> Result<Response, StdError> {
     let state = config_read(storage).load()?;
     if id >= state.next_swap_id {
         // FIXME(LR) >= ?
         return Err(StdError::generic_err(ERR_INVALID_SWAP_ID));
     }
     match refunds_have(id, storage) {
         true => Err(StdError::generic_err(ERR_ALREADY_REFUNDED)),
-        false => Ok(HandleResponse::default()),
+        false => Ok(Response::default()),
     }
 }
 
 /* ***************************************************
  * ************    Access Control      ***************
  * ***************************************************/
 
-fn only_admin<S: Storage, A: Api>(env: &Env, storage: &S, api: &A) -> HandleResult {
-    _only_role(&AccessRole::Admin, env, storage, api)
+fn only_admin(
+    info: &MessageInfo,
+    storage: &dyn Storage,
+    api: &dyn Api,
+) -> Result<Response, StdError> {
+    _only_role(&AccessRole::Admin, info, storage, api)
 }
 
-fn only_relayer<S: Storage, A: Api>(env: &Env, storage: &S, api: &A) -> HandleResult {
-    _only_role(&AccessRole::Relayer, env, storage, api)
+fn only_relayer(
+    info: &MessageInfo,
+    storage: &dyn Storage,
+    api: &dyn Api,
+) -> Result<Response, StdError> {
+    _only_role(&AccessRole::Relayer, info, storage, api)
 }
 
-fn only_approver<S: Storage, A: Api>(env: &Env, storage: &S, api: &A) -> HandleResult {
-    _only_role(&AccessRole::Approver, env, storage, api)
+fn only_approver(
+    info: &MessageInfo,
+    storage: &dyn Storage,
+    api: &dyn Api,
+) -> Result<Response, StdError> {
+    _only_role(&AccessRole::Approver, info, storage, api)
 }
 
-fn only_monitor<S: Storage, A: Api>(env: &Env, storage: &S, api: &A) -> HandleResult {
-    _only_role(&AccessRole::Monitor, env, storage, api)
+fn only_monitor(
+    info: &MessageInfo,
+    storage: &dyn Storage,
+    api: &dyn Api,
+) -> Result<Response, StdError> {
+    _only_role(&AccessRole::Monitor, info, storage, api)
 }
 
-fn _only_role<S: Storage, A: Api>(
+fn _only_role(
     role: &AccessRole,
-    env: &Env,
-    storage: &S,
-    api: &A,
-) -> HandleResult {
-    let env_message_sender = api.human_address(&env.message.sender)?;
+    info: &MessageInfo,
+    storage: &dyn Storage,
+    _api: &dyn Api,
```

Comment:
Maybe not as part of this PR but I guess we should remove this from this interface. This will probably have quite a bit of the ripple effect but I think that it will make things better in the long run

Reply:
Not that bad, see #44 :+1;